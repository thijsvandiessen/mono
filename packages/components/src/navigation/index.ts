export { Navigation } from './navigation.jsx'
