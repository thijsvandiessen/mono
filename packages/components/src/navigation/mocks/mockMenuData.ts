import type { GetGeneralInfoQuery } from '@mono/graphql/src/generated/graphql'

export const mockMenuData: GetGeneralInfoQuery = {
  general: {
    id: '187236701',
    title: 'algemene info',
    menu: [
      {
        id: '187236699',
        label: 'contact',
        link: {
          __typename: 'PageRecord',
          _createdAt: 'some date time string',
          _updatedAt: 'some date time string',
          id: 'unique-id',
          slug: 'contact',
        },
      },
      {
        id: '187236700',
        label: 'homepage',
        link: {
          __typename: 'PageRecord',
          _createdAt: 'some date time string',
          _updatedAt: 'some date time string',
          id: 'unique-id',
          slug: 'homepage',
        },
      },
      {
        id: '194164583',
        label: 'level 1',
        menu: [
          {
            id: '194164581',
            label: 'level 2 - contact',
            link: {
              __typename: 'PageRecord',
              _createdAt: 'some date time string',
              _updatedAt: 'some date time string',
              id: 'unique-id',
              slug: 'contact',
            },
          },
          {
            id: '194164582',
            label: 'level 2 - specific page',
            link: {
              __typename: 'PageRecord',
              _createdAt: 'some date time string',
              _updatedAt: 'some date time string',
              id: 'unique-id',
              slug: 'specific/page',
            },
          },
        ],
      },
    ],
  },
}
