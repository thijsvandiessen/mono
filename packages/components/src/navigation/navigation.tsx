import React from 'react'
import Link from 'next/link'
import classNames from 'classnames'
import { NavigationItem } from '../navigationItem'
import { NavigationSubMenu } from '../navigationSubMenu'
import { getGeneralInfo } from '@crea/graphql/src/getters/getGeneralInfo'
import { MobileMenu } from '../mobileMenu'
import { getSiteMetadata } from '@crea/graphql/src/getters/getSiteMetadata'

import styles from './styles.module.scss'
import { CreaOrkestLogo } from '../icons/crea-orkest-logo'

export const Navigation: React.FC = async () => {
  const { data } = await getGeneralInfo()
  const { metadata } = await getSiteMetadata()
  if (!data?.general) return null

  return (
    <nav className={classNames(styles.root, 'content-layout')}>
      <div className={classNames(styles.content)}>
        <Link href="/" className={classNames(styles.logoLink)}>
          <CreaOrkestLogo className={classNames(styles.logo)} />
          <span className="sr-only">{metadata?.title}: Ga naar home</span>
        </Link>
        <MobileMenu />
        <ul className={classNames(styles.list)}>
          {data.general.menu.map((item) => {
            if ('link' in item) {
              return (
                <NavigationItem
                  key={item.id}
                  slug={item?.link?.slug}
                  label={item.label}
                />
              )
            }

            if ('menu' in item) {
              return (
                <NavigationSubMenu
                  key={item.id}
                  label={item.label}
                  item={JSON.stringify(item.menu)}
                />
              )
            }

            return <NavigationItem key={item.id} slug={'/'} label={'Error'} />
          })}
        </ul>
      </div>
    </nav>
  )
}
