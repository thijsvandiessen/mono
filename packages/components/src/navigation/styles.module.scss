.root {
  background-color: var(--colour-background);
  margin-top: 2rem;
  margin-bottom: 2rem;
}

.content {
  display: flex;
  gap: 1rem;
  flex-direction: row;
  flex-wrap: wrap;

  @media screen and (min-width: 50rem) {
    flex-direction: row-reverse;
  }
}

.list {
  display: none;
  flex-direction: column;
  background-color: var(--colour-background);
  list-style: none;
  padding: 0;
  margin: 0;

  @media screen and (min-width: 50rem) {
    display: flex;
    flex-direction: row;
    align-items: center;
    gap: 3rem;
    flex-shrink: 0;
    margin-right: auto;
  }
}

.logoLink {
  margin: 1rem 0;
  margin-right: auto;

  @media screen and (min-width: 50rem) {
    margin-left: auto;
    margin-right: 0;
  }
}

.logo {
  position: relative;
  object-fit: contain;
  width: 175px;
  height: 40px;
}
