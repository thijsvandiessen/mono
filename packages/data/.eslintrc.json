{
  "ignorePatterns": ["**/generated/*.ts", "dist/*"],
  "plugins": ["@typescript-eslint", "testing-library"],
  "extends": [
    "eslint:recommended",
    "plugin:react-hooks/recommended",
    "plugin:@typescript-eslint/recommended"
  ],
  "parser": "@typescript-eslint/parser",
  "env": {
    "jest": true
  },
  "rules": {
    "require-await": ["warn"],
    "no-unused-vars": ["warn"],
    "@typescript-eslint/no-explicit-any": "warn",
    "@typescript-eslint/consistent-type-imports": "error",
    "sort-imports": [
      "warn",
      {
        "ignoreCase": false,
        "ignoreDeclarationSort": false,
        "ignoreMemberSort": false,
        "memberSyntaxSortOrder": ["none", "all", "multiple", "single"],
        "allowSeparatedGroups": false
      }
    ]
  },
  "root": true
}
