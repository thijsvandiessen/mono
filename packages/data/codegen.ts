import type { CodegenConfig } from '@graphql-codegen/cli'

const config: CodegenConfig = {
  schema: './src/generated/schema.graphql',
  documents: ['./**/*.ts'],
  emitLegacyCommonJSImports: false,
  generates: {
    './src/generated/': {
      preset: 'client',
      config: {
        withHooks: true,
        scalars: {
          BooleanType: 'boolean',
          CustomData: 'Record<string, string>',
          Date: 'string',
          DateTime: 'string',
          FloatType: 'number',
          IntType: 'number',
          ItemId: 'string',
          JsonField: 'unknown',
          MetaTagAttributes: 'Record<string, string>',
          UploadId: 'string',
        },
        useTypeImports: true,
        verbatimModuleSyntax: true,
      },
      presetConfig: {
        persistedDocuments: true,
        gqlTagName: 'gql',
        fragmentMasking: false,
      },
    },
    './src/generated/introspection.json': {
      plugins: ['introspection'],
      config: {
        minify: false,
        schemaDescription: true,
        federation: true,
      },
    },
  },
  hooks: { afterAllFileWrite: ['prettier --write'] },
  ignoreNoDocuments: true,
}

export default config
