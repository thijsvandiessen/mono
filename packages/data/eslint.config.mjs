import { defineConfig, globalIgnores } from 'eslint/config'
import { eslintConfig } from '@mono/eslint-config'

export default defineConfig([
  globalIgnores(['src/generated/*.ts', 'dist/*', 'node_modules']),
  eslintConfig,
])
