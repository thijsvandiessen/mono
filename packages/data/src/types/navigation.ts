export interface NavigationItem {
  id: string
  label?: string | null
  slug?: string | null
}

export interface NavigationSubMenu {
  id: string
  label?: string | null
  items: NavigationItem[]
}

export interface Navigation {
  id: string
  title?: string | null
  menu: Array<NavigationItem | NavigationSubMenu>
}
