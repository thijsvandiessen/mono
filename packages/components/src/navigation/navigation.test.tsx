import { describe, expect, it, vi } from 'vitest'
import { render, screen } from '@testing-library/react'
import { Navigation } from './navigation.js'
import React from 'react'
import { getGeneralInfo } from '@mono/data'
import { mockMenuData } from './mocks/mockMenuData.js'
import { resolvedComponent } from '@mono/utils'

global.ResizeObserver = vi.fn().mockImplementation(() => ({
  observe: vi.fn(),
  unobserve: vi.fn(),
  disconnect: vi.fn(),
}))

vi.mock('@mono/data', () => ({
  ...vi.importActual('@mono/data'),
  __esModule: true,
  client: vi.fn(),
  getGeneralInfo: vi.fn(),
  getSiteMetadata: vi.fn(() => ({
    metadata: {
      title: '',
      description: 'Default description',
      base_url: 'https://example.com',
    },
    error: undefined,
  })),
}))

const getGeneralInfoMock = vi.mocked(getGeneralInfo)

describe('Concert component', () => {
  it('shows all the data', async () => {
    getGeneralInfoMock.mockResolvedValueOnce({
      data: mockMenuData,
      error: undefined,
    })

    const Resolved = await resolvedComponent(Navigation, {})

    const { container } = render(<Resolved />)

    expect(container).toMatchSnapshot()
    expect(screen.getByText('contact')).toBeTruthy()
  })
})
