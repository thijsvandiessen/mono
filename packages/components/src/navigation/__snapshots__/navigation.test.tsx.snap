// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Concert component shows all the data 1`] = `
<div>
  <nav
    class="root content-layout"
  >
    <div
      class="content"
    >
      <a
        class="logoLink"
        href="/"
      >
        <svg
          aria-hidden="true"
          class="logo icon"
          height="64px"
          viewBox="0 0 276 64"
          width="276px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M176.41.66V33.8c1.88 0 3.71.01 5.54 0s3.68.1 5.49-.11c3.94-.46 6.49-2.85 7.33-6.79.34-1.58.35-3.24.64-4.83 1.08-5.77 4.07-9.97 10.01-11.53 3.2-.84 6.42-.71 9.57.35 2.42.82 4.39 2.27 5.94 4.31.18.24.36.47.57.75 3.78-5.29 9.07-6.65 15.09-5.61 6.41 1.11 9.7 5.43 11.03 11.57h21.49v5.06H247.9c-.01.27-.03.49-.03.71v20.23c0 1.83.44 3.57 1.26 5.21 1.37 2.72 3.67 4.17 6.65 4.37 1.55.11 3.19.03 4.69-.34 3.29-.82 4.97-3.27 5.77-6.4.2-.8.3-1.62.45-2.48h4.91c.13 4.07-2.07 10.55-7.85 13.08-3.26 1.43-6.65 1.63-10.1.94-3.5-.7-6.27-2.56-8.3-5.6-2.88 4.24-6.97 5.95-11.89 5.96-4.92 0-9.02-1.68-11.9-5.89-.08.07-.17.12-.22.19-2.58 3.73-6.26 5.44-10.68 5.68-3.38.19-6.61-.38-9.46-2.35-3.39-2.34-5.08-5.72-5.77-9.66-.28-1.57-.28-3.2-.63-4.76-1.01-4.53-4.04-6.93-8.69-6.97-3.06-.03-6.13 0-9.19-.01h-.49v21.2h-5.08V1.57c0-.3-.03-.6-.04-.9h5.12Zm47.61 47.58.03.46c.53 5.63 3.7 8.74 9.05 8.88 4.88.13 7.99-2.09 9.24-6.59.66-2.38.77-4.7-1.1-6.62-.97-.99-2.06-1.93-3.25-2.65-2.14-1.29-4.38-2.43-6.6-3.57-3.3-1.7-6.61-3.38-9.26-6.07-.02-.02-.09 0 .06 0-1 1.18-1.84 2.35-2.87 3.32-2.68 2.52-6.01 3.42-9.6 3.45-3.94.03-7.89 0-11.83 0h-.86c2.15 2.95 2.99 6.16 3.21 9.61.28 4.36 2.45 9.01 9.04 9.12 4.88.08 8.08-2.17 9.26-6.67.23-.86.33-1.76.5-2.68h4.97Zm-26.93-14.49c4.71 0 9.28.1 13.85-.04 2.82-.09 5.11-1.41 6.53-3.94 1.25-2.23 1.83-4.6 1.29-7.15-.73-3.43-2.36-6.17-5.98-7.1-1.38-.36-2.89-.44-4.32-.37-3.25.17-5.75 1.68-7.12 4.72-.45.99-.75 2.07-.95 3.14-.3 1.67-.37 3.39-.7 5.05-.41 2.04-1.31 3.91-2.61 5.69Zm45.71 5.12v-.98c0-4.26.01-8.52-.01-12.78 0-.85-.06-1.71-.24-2.54-.59-2.83-1.8-5.31-4.59-6.56-2.98-1.34-6.06-1.34-9.05-.01-2.74 1.21-3.92 3.61-4.61 6.38-.72 2.88.06 5.24 2.38 7 1.61 1.22 3.34 2.31 5.11 3.28 3.64 1.99 7.5 3.59 11.01 6.22ZM159.38 9.91c-4.03.52-7.28 2.36-9.6 5.83-1.82-2.71-4.21-4.48-7.25-5.35-2.22-.63-4.5-.78-6.79-.5-4.5.54-7.99 2.65-10.24 6.64-1.39 2.47-2.06 5.16-2.07 7.98-.01 3.96-.07 5.93-.02 22.79.04 1.02.15 2.18.41 3.45.49 2.4 1.33 4.34 2.09 5.75.48.66 1.16 1.45 2.06 2.29 3.23 3.02 7.18 3.84 11.44 3.52 4.23-.32 7.73-2.06 10.16-5.62 1.88-2.75 2.73-5.84 2.74-9.14.03-7.62 0-15.24.02-22.86 0-.7.06-1.4.17-2.09.79-4.95 3.96-7.74 8.97-7.74 1.5 0 3.01.32 4.6.5.57-1.34 1.22-2.85 1.86-4.34-1.92-1.03-5.49-1.5-8.54-1.11Zm-13.08 42.3c-1.29 3.06-3.63 4.77-6.91 5.01-1.58.12-3.24.04-4.77-.34-3.35-.84-5.02-3.37-5.78-6.56-.22-.92-.32-1.88-.33-2.82-.03-7.67-.01-15.34-.01-23.01 0-1.73.4-3.38 1.13-4.93 1.41-2.98 3.87-4.45 7.09-4.66 1.22-.08 2.48-.04 3.67.2 3.65.72 5.98 3.38 6.65 7.42.12.71.18 1.44.18 2.17.02 3.81 0 7.62 0 11.43v11.51c0 1.6-.33 3.13-.95 4.6Z"
            fill="#07161a"
            stroke-width="0"
          />
          <path
            d="M4.42 47.47c-.01.71-.03 1.43.09 2.43.2 1.68.99 5.96 4.63 9.12 3.87 3.37 8.48 3.95 13.3 2.97 5.73-1.17 9.62-5.57 10.53-11.63.17-1.11.24-2.24.26-3.37.04-2.78.01-5.55.01-8.39h.57c2.74 0 5.48-.03 8.21.01.97.02 1.96.1 2.91.33 3.55.83 6.09 2.8 6.81 6.55.25 1.32.21 2.7.38 4.03.5 3.79 1.9 7.16 4.88 9.7 3.06 2.61 6.73 3.35 10.62 3.12 4.39-.27 8.03-2.01 10.57-5.73.03-.05.12-.07.19-.11 2.87 4.19 6.97 5.88 11.89 5.87 4.91 0 9-1.72 11.88-5.94 2.89 4.24 6.99 5.94 11.9 5.95 1.64 0 4.94-.04 8.21-2.21 1.86-1.24 3.02-2.7 3.66-3.66-.51-.95-1.05-2.1-1.49-3.45-.7-2.15-.95-4.08-1.02-5.56-.05 1.67-.4 3-.7 3.88-.41 1.2-1.03 2.96-2.68 4.2-.89.69-1.97 1.19-3.28 1.46-1.36.28-2.8.32-4.19.21-3.27-.27-5.63-1.94-6.92-5-.64-1.51-.97-3.09-.96-4.75.01-7.65.03-15.29 0-22.94-.01-3.22-.85-6.25-2.65-8.95-2.46-3.69-6.03-5.45-10.39-5.76-4.36-.31-8.34.6-11.56 3.77-.63.63-1.17 1.35-1.8 2.07-2.85-4.2-6.96-5.89-11.86-5.9-4.93 0-9.06 1.71-11.9 5.92-.47-.58-.85-1.11-1.28-1.59-2.54-2.83-5.82-4.11-9.54-4.29-5.14-.26-9.59 1.14-12.7 5.52-1.93 2.72-2.81 5.81-2.83 9.11-.04 7.62 0 15.24-.03 22.86 0 .99-.1 2.01-.33 2.97-.7 2.99-2.2 5.41-5.26 6.43-1.19.39-2.5.56-3.76.59-3.78.08-6.75-1.32-8.36-4.94-.68-1.53-1.02-3.13-1.02-4.81 0-7.57 0-15.14.02-22.71 0-1.04.13-2.1.35-3.12.88-3.92 3.3-6.19 7.28-6.77 2.32-.34 4.61-.11 6.78 1.03.84-1.47 1.66-2.91 2.49-4.38-3.26-1.74-6.65-2.15-10.16-1.62-1.52.23-4.18.75-6.73 2.83-3.19 2.61-4.15 6.05-4.53 7.52-.43 1.64-.47 2.87-.51 4.55m46 4.76c-1.34 2.35-3.5 3.66-6.14 3.79-3.65.18-7.31.04-11.04.04v-3.12c0-1.98 0-3.96.01-5.94 0-1.68.39-3.27 1.09-4.79 1.4-3.02 3.87-4.52 7.12-4.72 1.24-.08 2.52-.03 3.74.21 3.69.73 5.57 3.27 6.46 6.73.71 2.78.14 5.37-1.24 7.8Zm6.7-5.77c.11-1.46.52-2.97 1.11-4.32 1.32-3.02 3.85-4.45 7.07-4.65 1.22-.08 2.48-.04 3.67.2 3.67.74 6 3.44 6.67 7.5.44 2.68-.2 5.12-1.61 7.38-1.38 2.2-3.51 3.39-6.02 3.47-4.66.15-9.32.04-14.07.04 2.11-2.91 2.92-6.15 3.19-9.62Zm19.13 20.86c-.32 1.56-.33 3.19-.64 4.75-.22 1.14-.57 2.29-1.06 3.34-1.3 2.75-3.63 4.13-6.6 4.44-1.81.19-3.61.12-5.35-.52-2.97-1.1-4.5-3.42-5.18-6.34-.3-1.28-.34-2.62-.45-3.94-.23-2.78-1.06-5.33-2.81-7.55-.04-.05-.06-.12-.13-.28h12.3c1.97 0 3.91-.25 5.76-.96 4.46-1.71 7.05-5.08 8.18-9.59.44-1.78.48-3.67.8-5.48.5-2.85 1.67-5.37 4.41-6.69 3.13-1.51 6.38-1.52 9.51-.02 2.76 1.32 3.88 3.85 4.43 6.68.14.73.18 1.49.18 2.24.02 2.9 0 5.8 0 8.76h-3.21c-2.49.02-4.98-.08-7.45.09-6.71.45-11.33 4.5-12.7 11.08Zm23.14 4.83c-.33 3.27-1.97 5.96-5.29 7.15-2.58.93-5.23.92-7.81-.04-2.91-1.08-4.31-3.43-5.07-6.27-.68-2.55-.24-4.97.91-7.28 1.46-2.94 3.98-4.34 7.15-4.45 3.41-.12 6.82-.03 10.43-.03-.09 3.68.05 7.33-.32 10.92Z"
            fill="#60acbd"
            stroke-width="0"
          />
        </svg>
        <span
          class="sr-only"
        >
          Default title
          : Ga naar home
        </span>
      </a>
      <div
        class="root"
      >
        <button
          class="menuButton"
          type="button"
        >
          <span
            class="sr-only"
          >
            Menu
          </span>
          <svg
            aria-hidden="true"
            class="menuIcon icon"
            height="24px"
            viewBox="0 0 24 24"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21.82 6.18H2.18C.98 6.18 0 5.2 0 4c0-1.2.98-2.18 2.18-2.18h19.64C23.02 1.82 24 2.8 24 4c0 1.2-.98 2.18-2.18 2.18zM21.82 22.18H2.18C.98 22.18 0 21.2 0 20c0-1.2.98-2.18 2.18-2.18h19.64c1.2 0 2.18.98 2.18 2.18 0 1.2-.98 2.18-2.18 2.18zM21.82 14.18H2.18C.98 14.18 0 13.2 0 12c0-1.2.98-2.18 2.18-2.18h19.64c1.2 0 2.18.98 2.18 2.18 0 1.2-.98 2.18-2.18 2.18z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
      <ul
        class="list"
      >
        <li
          class="navItem"
        >
          <a
            class="link text-large"
            href="/contact"
          >
            contact
          </a>
        </li>
        <li
          class="navItem"
        >
          <a
            class="link text-large"
            href="/"
          >
            homepage
          </a>
        </li>
        <li
          class="listItem"
        >
          <button
            class="button text-large"
            type="button"
          >
            level 1
          </button>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
